File: app/src/main/java/com/martinlaizg/geofind/data/access/database/dao/PlaceDAO.java
package com.martinlaizg.geofind.data.access.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.martinlaizg.geofind.data.access.database.entities.Place;

import java.util.List;

@Dao
public interface PlaceDAO {

	@Insert(onConflict = OnConflictStrategy.REPLACE)
	void insert(Place place);

	@Update
	void update(Place place);

	@Query("SELECT * FROM places WHERE id = :placeId")
	Place getPlace(int placeId);

	@Query("SELECT * FROM places WHERE tour_id = :tour_id")
	List<Place> getTourPlaces(Integer tour_id);
}

File: app/src/main/java/com/martinlaizg/geofind/data/access/database/dao/PlaceDAO.kt
package com.martinlaizg.geofind.data.access.database.dao

import androidx.room.*
import com.martinlaizg.geofind.data.access.database.entities.Place

@Dao
interface PlaceDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(place: Place?)

    @Update
    fun update(place: Place?)

    @Query("SELECT * FROM places WHERE id = :placeId")
    fun getPlace(placeId: Int): Place?

    @Query("SELECT * FROM places WHERE tour_id = :tourId ORDER BY `order` ASC")
    fun getTourPlaces(tourId: Int?): MutableList<Place?>?
}